stdlib/public/core/OutputStream.swift

```diff
@@ -330,9 +330,19 @@ internal struct _Stdout : OutputStreamType {
   }
 
   mutating func write(string: String) {
-    // FIXME: buffering?
-    // It is important that we use stdio routines in order to correctly
-    // interoperate with stdio buffering.
+    if string.isEmpty { return }
+
+    if string._core.isASCII {
+      defer { _fixLifetime(string) }
+
+      let result = _swift_stdlib_fwrite_stdout(
+        UnsafePointer(string._core.startASCII), string._core.count, 1)
+      if result != 1 {
+        fatalError("fwrite() returned \(result), expected 1")
```

Comment:
Here's an idea.  Shall we commit this PR without the `fatalAssert()` - so behaviour isn't changed?  After all, the PR is about using `fwrite()`.

If you want to add the `fatalAssert()` I am very happy to do a separate PR immediately.  That way, if the change in error-handling behaviour causes any problems we can revert the second PR easily without affecting the `fwrite()` improvement.

@gribozavr - how does that sound?